internal/service/encryptionatrestprivateendpoint/resource.go

```diff
@@ -64,8 +69,17 @@ func (r *encryptionAtRestPrivateEndpointRS) Create(ctx context.Context, req reso
 
 	privateEndpointModel := NewTFEarPrivateEndpoint(*finalResp, projectID)
 	resp.Diagnostics.Append(resp.State.Set(ctx, privateEndpointModel)...)
-	if err := getErrorMsgForFailedStatus(finalResp); err != nil {
-		resp.Diagnostics.AddError(failedStatusErrorMessage, err.Error())
+
+	if shouldError, msgSummary, errMsg := CheckErrorMessageAndStatus(finalResp); errMsg != nil || shouldError {
+		if shouldError {
+			resp.Diagnostics.AddError(*msgSummary, *errMsg)
+			return
+		}
+		resp.Diagnostics.AddWarning(*msgSummary, *errMsg)
+		return
+	}
+	if isStatusPendingAcceptance(finalResp) {
+		resp.Diagnostics.AddWarning(pendingAcceptanceWarnMsgSummary, pendingAcceptanceWarnMsg)
```

Comment:
seeing more logic is added, WDYT about defining the method as `CheckErrorMessageAndStatus(model *admin.EARPrivateEndpoint) diag.Diagnostics` and unify logic for handling error vs warning + pending acceptance warning in same place?

Reply:
I did consider that initially, but then decided to keep it separate, mostly because CheckErrorMessageAndStatus focusses more on handling error scenarios whereas `if isStatusPendingAcceptance()` block is adding more of a generic/informative warning.